app/models/user.rb

```diff
@@ -96,7 +96,7 @@ def self.from_omniauth(auth)
 
   def self.from_mail_identifier(mail_id)
     all.each do |user|
-      return user if user.email.split('@').first.casecmp(mail_id).zero?
+      return user if user.human_readable_identifier.casecmp(mail_id).zero?
```

Comment:
definitely keep the change to use the human_readable_identifier here and in the other places though, it is a lot cleaner than before 👍